### PR TITLE
feat: add blog post about consistent URLs in Gatsby

### PR DESCRIPTION
**Pull request by feedm3**
As we recently spend some time improving our URLs for Google, I decided to write a blog post about what we have learned.

Preview URL: https://satellytescommain-featblogpostgatsbyurl.gtsb.io/blog/how-consistency-helps-you-to-optimize-gatsby-urls/

### CODE REVIEW

File: blog-posts/optimze-gatsby-urls.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+- [https://satellytes.com/career/325420-senior-frontend-engineer-(mwx)/](https://satellytes.com/career/325420-senior-frontend-engineer-(mwx)/)
+- [https://satellytes.com/career/325420-senior-frontend-engineer-(mwx)/index.html](https://satellytes.com/career/325420-senior-frontend-engineer-(mwx)/index.html)
+- all 3 links above, but starting with `https://www.`
+- all 3 links above, but starting with `http://`
+- all 3 links above, but starting with `http://www.`
+
+In general, it's good that all links are supported, as it lowers the possibility of a wrong URL entered by the user. But it also comes with 2 downsides:
+
+1. Degraded performance for the user as only one link returns the HTML file directly. All other links are redirects.
+2. Duplicate content for search engines. A page should only have one URL (or at least one "original" URL)
```
Comment (georgiee) [inline blog-posts/optimze-gatsby-urls.md]:
```suggestion
In general, it's good that all links are supported, as it lowers the possibility of a wrong URL entered by the user. But it also comes with downsides:

1. We are lost in the freedom of choice and randomly pick any url format because all of them are working.
2. Degraded performance for the user as only one link returns the HTML file directly. All other links are redirects.
3. Duplicate content for search engines. A page should only have one URL (or at least one "original" URL)
```

I added one item to explain the overall problem and how we arrived there. We never decided for a format, netlify was forgiving and everyone picked a random url format becaus it just worked.

File: blog-posts/optimze-gatsby-urls.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
+---
+path: "/blog/how-to-optimize-gatsby-urls/"
+date: "2021-07-15"
+title: "How consistency helps you to optimize Gatsby URLs for users and search engines"
+featuredImage: images/how-to-optimize-gatsby-url-hero.jpeg
+author: "Fabian"
+authorSummary: "Senior Developer at Satellytes"
+attribution:
+  creator: Jonny Clow
+  source: https://unsplash.com/photos/FwMBtl6IQQA
```
Comment (georgiee) [inline blog-posts/optimze-gatsby-urls.md]: Thanks for the attribution 👍

File: blog-posts/optimze-gatsby-urls.md [hunk 3]
```diff
@@ -0,0 +1,153 @@
+---
+path: "/blog/how-to-optimize-gatsby-urls/"
+date: "2021-07-15"
+title: "How consistency helps you to optimize Gatsby URLs for users and search engines"
```
Comment (georgiee) [inline blog-posts/optimze-gatsby-urls.md]: path and title are different. We might want to create the path/slug automatically from our title in the future? Because this creates, wait for it... inconsistency in urls. Some match the title, some are not 😅

File: blog-posts/optimze-gatsby-urls.md [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+featuredImage: images/how-to-optimize-gatsby-url-hero.jpeg
+author: "Fabian"
+authorSummary: "Senior Developer at Satellytes"
+attribution:
+  creator: Jonny Clow
+  source: https://unsplash.com/photos/FwMBtl6IQQA
+---
+
+We did a deep dive into how to work with URLs in Gatsby to improve UX and search engine ranking. See what we have
```
Comment (georgiee) [inline blog-posts/optimze-gatsby-urls.md]: Perfect intro and makes me want to read the article 👌

File: blog-posts/optimze-gatsby-urls.md [hunk 5]
```diff
@@ -0,0 +1,153 @@
+---
+path: "/blog/how-to-optimize-gatsby-urls/"
+date: "2021-07-15"
```
Comment (georgiee) [inline blog-posts/optimze-gatsby-urls.md:3]: date is good (classic mistake :D )👍

File: blog-posts/optimze-gatsby-urls.md [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+    >
+      {children}
+    </GatsbyLink>
+  );
+};
+```
+
+> Check out the full component implementation on our public website repository: [https://github.com/satellytes/satellytes.com/blob/main/src/components/links/links.tsx](https://github.com/satellytes/satellytes.com/blob/main/src/components/links/links.tsx)
+
+With this component, we get consistency into our URLs (inconsistent URLs could come from authors or developers by accident). The trailing slash will be appended to all internal links and users don't have to deal with redirects - at least when they open the link in a new tab or share it to their friends, as client-side routing with `GatsbyLink` is not affected by this. Having all URLs aligned also helps us with SEO, which we go into detail in the next paragraph.
```
Comment (georgiee) [inline blog-posts/optimze-gatsby-urls.md]: I have a hard time understanding this part. 
I get the meaning (client side doesn't cause any trouble anyway, it's for the URL people might share from the address bar). It's super hard to follow.

> The trailing slash will be appended to all internal links and users don't have to deal with redirects - at least when they open the link in a new tab or share it to their friends, as client-side routing with `GatsbyLink` is not affected by this.

Comment (georgiee) [inline blog-posts/optimze-gatsby-urls.md]: In addition the first sentence is also difficult to understand with the parentheses. I proposed a slightly different sentence before, but you probably did not like it :D 

Here the same sentence with the addition about authoring wrong content. Maybe it helps to rephrase?

> This link component enforces a consistent format for any given URL if someone accidentally decided for a url without a trailing slash

File: blog-posts/optimze-gatsby-urls.md [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+
+With the canonical URL in the header, it's not a problem anymore for search engines that our pages have multiple URLs to reach them. As Netlify does a redirect for all URLs except one, this should be a problem in the first place. But depending on the webserver, this isn't always the case and although the redirects are correct in our case, the Google Search Console marked some URLs as duplicated content. You can read more about this topic on the Google documentation: [https://developers.google.com/search/docs/advanced/guidelines/duplicate-content](https://developers.google.com/search/docs/advanced/guidelines/duplicate-content)
+
+## Conclusion
+
+Consistency in URLs is important for UX and search engine rankings. To improve the consistency:
+
+1. Decide for one format - usually, the best format is given by the webserver or hoster
+2. Create a common `Link` component to ensure the formatting of internal links on your website
+3. Align all `createPage` function calls for a correct `sitemap.xml` and canonical URL
```
Comment (georgiee) [inline blog-posts/optimze-gatsby-urls.md]: > The freedom of choice given by the tools we use is sometimes difficult and has implications that are not obvious. > Follow these simple rules in your Gatsby application and you will create uniform links for a better SEO and UX.

**Review by georgiee — APPROVED** [review]
I have a few comments and additions. Feel free to apply them.
Great blog post and I'm looking forward to have them on our blog ✨

File: blog-posts/optimze-gatsby-urls.md (hunk 3 above)
Comment (feedm3) [reply blog-posts/optimze-gatsby-urls.md]: Oh no, not again 😄 I changed the url, it's now like the title

File: blog-posts/optimze-gatsby-urls.md (hunk 6 above)
Comment (feedm3) [reply blog-posts/optimze-gatsby-urls.md]: I changed this part:

> With this component, we get consistency into our URLs as the component makes sure that every link gets rendered with a trailing slash. The means the users don't have to deal with redirects anymore - at least when they open the link in a new tab or share it to their friends, as client-side routing with `GatsbyLink` is not affected by this. Having all URLs aligned also helps us with SEO, which we go into detail in the next paragraph.

File: blog-posts/optimze-gatsby-urls.md (hunk 1 above)
Comment (feedm3) [reply blog-posts/optimze-gatsby-urls.md]: I applied your point, good one!

File: blog-posts/optimze-gatsby-urls.md (hunk 6 above)
Comment (feedm3) [reply blog-posts/optimze-gatsby-urls.md]: I removed the sentence in the parenthesis. I put the sentence into the paragraph above the code example:

> That's why we should make sure to always render the direct URL on our page, even if the developers or authors put in the wrong one.

File: blog-posts/optimze-gatsby-urls.md (hunk 7 above)
Comment (feedm3) [reply blog-posts/optimze-gatsby-urls.md]: > The freedom of choice given by the tools we use is sometimes difficult and has implications that > are not obvious. Follow these simple rules in your Gatsby application and you will create uniform links for a better SEO and UX:
>
> 1. Decide for one URL format - usually, the best format is given by the webserver or hoster
> 2. Create a common `Link` component to ensure the URL format of internal links rendered on your website
> 3. Align the URL format in all of your `createPage` function calls for a correct `sitemap.xml` and canonical URL
